quicklogic/common/cmake/quicklogic_install.cmake

```diff
@@ -116,45 +82,28 @@ function(DEFINE_QL_TOOLCHAIN_TARGET)
           PERMISSIONS WORLD_READ OWNER_WRITE OWNER_READ GROUP_READ)
 
   # install techmap
-  install(DIRECTORY ${symbiflow-arch-defs_SOURCE_DIR}/quicklogic/${FAMILY}/techmap
-      DESTINATION share/techmaps/quicklogic/${FAMILY})
+  install(DIRECTORY ${symbiflow-arch-defs_SOURCE_DIR}/quicklogic/${FAMILY}/techmap/.
+          DESTINATION share/symbiflow/techmaps/${FAMILY}
+          FILES_MATCHING PATTERN *.v)
 
   install(FILES ${DEFINE_QL_TOOLCHAIN_TARGET_CELLS_SIM}
-          DESTINATION share/techmaps/quicklogic/${FAMILY}/techmap)
+          DESTINATION share/symbiflow/techmaps/${FAMILY})
 
   # install Yosys scripts
-  install(FILES  ${DEFINE_QL_TOOLCHAIN_TARGET_CONV_SCRIPT} ${DEFINE_QL_TOOLCHAIN_TARGET_SYNTH_SCRIPT}
-    DESTINATION share/quicklogic/${FAMILY}/yosys)
-
-  # Install PP3 specific files
-  if("${FAMILY}" STREQUAL "pp3")
-
-    # Database
-    install(FILES ${symbiflow-arch-defs_SOURCE_DIR}/build/quicklogic/${FAMILY}/devices/ql-eos-s3-virt/db_phy.pickle
-      DESTINATION share/arch/ql-eos-s3_wlcsp
-      PERMISSIONS WORLD_READ OWNER_WRITE OWNER_READ GROUP_READ )
-
-    # Python scripts
-    install(FILES ${symbiflow-arch-defs_SOURCE_DIR}/quicklogic/common/utils/eos_s3_iomux_config.py
-      DESTINATION bin/python
-      PERMISSIONS WORLD_EXECUTE WORLD_READ OWNER_WRITE OWNER_READ OWNER_EXECUTE GROUP_READ GROUP_EXECUTE)
-
-    # Yosys scripts
-    install(FILES ${symbiflow-arch-defs_SOURCE_DIR}/quicklogic/${FAMILY}/yosys/pack.tcl
-      DESTINATION share/quicklogic/${FAMILY}/yosys
-      PERMISSIONS WORLD_READ OWNER_WRITE OWNER_READ GROUP_READ)
-
-  # Install AP3 specific files
-  elseif("${FAMILY}" STREQUAL "ap3")
+  install(FILES ${DEFINE_QL_TOOLCHAIN_TARGET_CONV_SCRIPT} ${DEFINE_QL_TOOLCHAIN_TARGET_SYNTH_SCRIPT}
+          DESTINATION share/symbiflow/scripts/${FAMILY})
 
-    # Anything?
-
-  # Unsupported family
-  else()
-    message(FATA_ERROR "Family '${FAMILY}' is not suitable for installed toolchain")
-  endif()
+  install(FILES ${symbiflow-arch-defs_SOURCE_DIR}/quicklogic/${FAMILY}/utils/create_ioplace.py
+          DESTINATION bin/python
+          PERMISSIONS WORLD_EXECUTE WORLD_READ OWNER_WRITE OWNER_READ OWNER_EXECUTE GROUP_READ GROUP_EXECUTE)
 
+  install(FILES ${symbiflow-arch-defs_SOURCE_DIR}/quicklogic/${FAMILY}/devices/umc22/interface-mapping_24x24.xml
+          DESTINATION share/symbiflow/arch/qlf_k4n8-qlf_k4n8_umc22_qlf_k4n8-qlf_k4n8_umc22
```

Comment:
can't this filename and install be somehow calculated? hardcoding it see doesn't feel right (this file is common to the whole QL flow)

Reply:
Right. I see that this file belongs actually to a device not the arch and is now being installed with the arch. Need to fix that to be more generic.

Reply:
Fixed.